src/util/getCommandState.ts

```diff
@@ -87,6 +89,25 @@ const getCommandState = (value: string): CommandState => {
       matches[command] &&= currentCommandState[command]
     }
   }
+  if (savedValue.length > 0) {
+    const parser = new DOMParser() // new DomParser
```

Comment:
`getCommandState` is called on every keystroke, so it's highly performance sensitive. Instantiating a new `DOMParser` on every keystroke is too heavy a solution unfortunately.

We could maybe use this approach if we throttle it.

Reply:
Ok, I will check about that.

Reply:
`if (savedValue.length > 0 && (savedValue.match(fontRegex) || savedValue.match(spanRegex)))`

I think if we use this, it will be solved.
**DOMParser** will be created when needed.

Reply:
That helps, but the problem still exists when color is used. We either need a very efficient parser that is tailor-made for parsing color, or we need to throttle this at least to the next animation frame.

Reply:
@raineorshine , I have fixed getCommandState function without using **DOMParser**.
Please check and let me know.

Reply:
@raineorshine , What do you think about the current getCommandState function?
I removed DOMParser.

Reply:
Will take a look tomorrow! I was at a conference this week.

Reply:
Ok, I got it.

Reply:
That looks much better. If we run into performance issues, then it's a problem with the parsing logic of `getCommandQuery` and can be addressed in a separate issue. This isn't adding major performance overhead any more.

Reply:
Hello @raineorshine , When will you check the current functionalities?

Reply:
https://github.com/cybersemics/em/pull/2289#pullrequestreview-2245841057